core-bundle/src/Image/Preview/PreviewProviderInterface.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Image\Preview;
+
+use Contao\Image\ImageDimensions;
+
+interface PreviewProviderInterface
+{
+    public function supports(string $path, string $fileHeader = ''): bool;
+
+    /**
+     * @param int $size Minimal size the preview image should have in each
+     *                  dimension. Can be larger for implementation reasons or
+     *                  smaller if there is not enough data.
+     */
+    public function generatePreview(string $sourcePath, int $size, string $targetPath): void;
```

Comment:
What if generating fails?

Reply:
Will create a custom exception for that and add an `@throws` here.

Reply:
80b270b1b0084668baa8c02c2d33eac648208a28